pynest/examples/glif_psc_neuron.py

```diff
@@ -0,0 +1,322 @@
+# -*- coding: utf-8 -*-
+#
+# glif_psc_neuron.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Current-based generalized leaky integrate and fire (GLIF) neuron example
+--------------------------------
+
+Simple example of how to use the ``glif_psc`` neuron model for
+five different levels of GLIF neurons.
+
+Four stimulation paradigms are illustrated for the GLIF model
+with externally applied current and spikes impinging
+
+Voltage traces, current traces, threshold traces, and spikes are shown.
+
+KEYWORDS: glif_psc
+"""
+
+##############################################################################
+# First, we import all necessary modules to simulate, analyze and plot this
+# example.
+
+import nest
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+
+##############################################################################
+# We initialize the nest and set the simulation resolution.
+
+nest.ResetKernel()
+resolution = 0.05
+nest.SetKernelStatus({"resolution": resolution})
+
+##############################################################################
+# We define functions for creation of the five GLIF models
+# using the GLIF parameters of cell 490626718 in Allen Cell Type Database
+# (ACTD). The parameters is available at
+# <https://celltypes.brain-map.org/mouse/experiment/electrophysiology/490626718>
+# with units being converted from SI units( i.e., V, S( 1 / Ohm ), F, s, A ) to
+# NEST used units( i.e., mV, nS( 1 / GOhm ), pF, ms, pA ) and values being
+# rounded to appropriate digits for simplification.
+# Note that the rounded values of the Level 5 GLIF model of such a cell are
+# the default parameter values of ``glif_psc`` neuron.
+# Each function includes effected parameters for each level of GLIF model.
+# Also, the initial membrane potential V_m is set to E_L.
+
+
+def create_lif_psc(mechs, syn_tau):
```

Comment:
The `mechs` parameter makes no sense: Since this function must create a LIF model, you require that `mechs==[False, False, False]`. If it is not, you get a different model back, which would be very confusing. The same holds for the other `create_...()` functions.

Also, it seems as if you set almost every parameter here. Is that strictly necessary? The example should introduce the user to the model, so all the code setting parameters may distract. Would it be possible to just use the default parameters?

Maybe then one could drop the `create_...()` functions entirely?

Reply:
@heplesser The reason for the `create_...()` function is that users can see what parameters are associated with the selected level of models. All the parameters are repeating the default parameters in the model. Do you think this is necessary to help user get familiar with the parameter set of each level of glif model? If so, I will remove such function and change the example to the one you mentioned in the next comment.

Reply:
Examples have been updated.